src/oss-find-squats-lib/SquatChecker.cs

```diff
@@ -0,0 +1,119 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// --------------------------------------------------------------------------------------------
+
+using Microsoft.CST.OpenSource.Model.Mutators;
+using Microsoft.CST.OpenSource.Shared;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace Microsoft.CST.OpenSource.FindSquats
+{
+    /// <summary>
+    /// The class to wrap around the mutators to get the dictionary of potential package mutations
+    /// instead of printing them to the console as is done in <see cref="FindSquatsTool"/>.
+    /// </summary>
+    public class SquatChecker
+    {
+        /// <summary>
+        /// The options for <see cref="SquatChecker"/>.
+        /// </summary>
+        public class Options
+        {
+            public int SleepDelay { get; set; }
+        }
+
+        public static NLog.ILogger Logger { get; set; } = NLog.LogManager.GetCurrentClassLogger();
+
+
+        /// <summary>
+        /// Generate the mutations for the package.
+        /// </summary>
+        /// <returns>A dictionary of each mutation, with a list of the reasons why this mutation showed up.</returns>
+        /// <exception cref="InvalidOperationException">Thrown if the package doesn't have a name.</exception>
+        public Dictionary<string, IList<string>> Mutate(PackageURL pUrl, BaseProjectManager manager)
+        {
+            Dictionary<string, IList<string>> mutations = new();
+
+            // Go through each mutator in this package manager and generate the mutations.
+            var mutationsList = manager.Mutators.SelectMany(m =>
+                m.Generate(pUrl.Name ?? throw new InvalidOperationException()));
```

Comment:
Can we do the null check outside of this `Generate` call? In my opinion it's a little cleaner to handle it at the beginning of the function and then continue with the confirmation that the name is valid.